tests/unit/test_harvested_paper.py

```
@@ -0,0 +1,328 @@
+        paper1 = HarvestedPaper(title="Deep Learning", source=HarvestSource.ARXIV)
+        paper2 = HarvestedPaper(title="DEEP LEARNING", source=HarvestSource.ARXIV)
+        paper3 = HarvestedPaper(title="deep learning", source=HarvestSource.ARXIV)
+
+        assert paper1.compute_title_hash() == paper2.compute_title_hash()
+        assert paper2.compute_title_hash() == paper3.compute_title_hash()
+
+    def test_compute_title_hash_ignores_punctuation(self):
+        """Title hash ignores punctuation."""
+        paper1 = HarvestedPaper(title="Deep Learning", source=HarvestSource.ARXIV)
+        paper2 = HarvestedPaper(title="Deep, Learning!", source=HarvestSource.ARXIV)
+        paper3 = HarvestedPaper(title="Deep-Learning?", source=HarvestSource.ARXIV)
```

Ruff (0.15.0)

[error] Local variable `paper3` is assigned to but never used

Remove assignment to unused variable `paper3`

(F841)